rules/doctrine-code-quality/src/Rector/Property/RemoveRedundantDefaultAnnotationValuesRector.php

```diff
@@ -0,0 +1,268 @@
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorJoinColumnAnnotation(Node $node): void
+    {
+        $joinColumnTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineJoinColumnTagValueNode($node);
+        if ($joinColumnTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($joinColumnTagValueNode, 'nullable', true);
+        $this->removeItemWithDefaultValue($joinColumnTagValueNode, 'referencedColumnName', 'id');
+        $this->removeItemWithDefaultValue($joinColumnTagValueNode, 'unique', false);
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorGeneratedValueAnnotation(Node $node): void
+    {
+        $generatedValue = $this->doctrinePropertyAnalyzer->matchDoctrineGeneratedValueTagValueNode($node);
+        if ($generatedValue === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($generatedValue, 'strategy', 'AUTO');
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorManyToManyAnnotation(Node $node): void
+    {
+        $manyToManyTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineManyToManyTagValueNode($node);
+        if ($manyToManyTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($manyToManyTagValueNode, 'orphanRemoval', false);
+        $this->removeItemWithDefaultValue($manyToManyTagValueNode, 'fetch', 'LAZY');
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorManyToOneAnnotation(Node $node): void
+    {
+        $manyToOneTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineManyToOneTagValueNode($node);
+        if ($manyToOneTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($manyToOneTagValueNode, 'fetch', 'LAZY');
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorOneToManyAnnotation(Node $node): void
+    {
+        $oneToManyTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineOneToManyTagValueNode($node);
+        if ($oneToManyTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($oneToManyTagValueNode, 'orphanRemoval', false);
+        $this->removeItemWithDefaultValue($oneToManyTagValueNode, 'fetch', 'LAZY');
+    }
+
+    /**
+     * @param Property $node
+     */
+    private function refactorOneToOneAnnotation(Node $node): void
+    {
+        $oneToManyTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineOneToOneTagValueNode($node);
+        if ($oneToManyTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($oneToManyTagValueNode, 'orphanRemoval', false);
+        $this->removeItemWithDefaultValue($oneToManyTagValueNode, 'fetch', 'LAZY');
+    }
+
+    /**
+     * @param bool|string|int $defaultValue
```

Comment:
I don't like this too much but found it as the best solution to not repeat the same code over and over again.